source/js/dismissableNotices.ts

```diff
@@ -0,0 +1,113 @@
+enum NoticeTimeout {
+  Session = 'session',
+  Permanent = 'permanent',
+  Imidiate = 'imidiate',
+}
+
+class DismissableNotice {
+  private notice: HTMLElement;
+  private dismissTrigger: HTMLElement | null;
+  private uid: string;
+  private timeout: NoticeTimeout;
+
+  constructor(notice: HTMLElement) {
+      this.notice = notice;
+      this.dismissTrigger = this.notice.querySelector('[data-dismissable-notice-trigger="1"]');
+      this.uid = this.notice.getAttribute('data-dismissable-notice-uid') || '';
+      this.timeout = this.getTimeoutValue();
+
+      // If the notice has no UID and is not set to be dismissed immediately,
+      if (!this.uid && this.timeout !== NoticeTimeout.Imidiate) {
```

Comment:
if the UID is missing we return here. Wouldnt it be frustrating for the user if the notice is dismissable but has no uid and cant be dismissed. Maybe dismiss until page is reloaded?